src/Notification.ts

```diff
@@ -11,11 +11,11 @@ export class Notification<T> {
   observe(observer: Observer<T>): any {
```

Comment:
type definition still requires `observer` interface, so typescript won't be able to pass partial implementation of observers. Should it be updated in this PR, or leave it for further type enhancement?